### PR TITLE
VZ-8114: Pod security for Jaeger Operator and Jaeger.

### PR DESCRIPTION
**Pull request by pshirodkar-orcl**
Pod security for Jaeger Operator and Jaeger.

Verrazzano currently uses Jaeger 1.37.0. Custom Resource Definition for Jaeger 1.37.0 and below does not support container security context settings. Helm charts for Jaeger Operator updated for container security context.

### CODE REVIEW

File: tests/e2e/verify-install/security/pod_security_test.go [hunk 1]
```diff
@@ -42,15 +42,12 @@ var skipPods = map[string][]string{
...
-	"verrazzano-monitoring": {
-		"jaeger",
-	},
 	"verrazzano-backup": {
 		"restic",
 	},
 }
 
-var skipContainers = []string{}
+var skipContainers = []string{"jaeger-collector", "jaeger-query", "jaeger-agent"}
```
Comment (pshirodkar-orcl) [inline tests/e2e/verify-install/security/pod_security_test.go:50]: Once we uptake Jaeger 1.38 and above in the future, we can specify container security context for these 3 containers in Jaeger Custom Resource which will allow us to remove the check from being skipped.

File: platform-operator/helm_config/overrides/jaeger-operator-values.yaml [hunk 2]
```diff
@@ -27,6 +46,29 @@ jaeger:
...
       options:
         es:
-          index-prefix: verrazzano
\ No newline at end of file
+          index-prefix: verrazzano
+
+securityContext:
+  runAsGroup: 65534
+  runAsNonRoot: true
+  runAsUser: 1001
```
Comment (pfmackin) [inline platform-operator/helm_config/overrides/jaeger-operator-values.yaml:62]: shouldn't the group be 65534

File: platform-operator/helm_config/overrides/jaeger-operator-values.yaml [hunk 3]
```diff
@@ -27,6 +46,29 @@ jaeger:
...
+  seccompProfile:
+    type: RuntimeDefault
+
+containerSecurityContext:
+  runAsUser: 1001
```
Comment (pfmackin) [inline platform-operator/helm_config/overrides/jaeger-operator-values.yaml:67]: shouldn't the group be 65534

File: platform-operator/helm_config/overrides/jaeger-operator-values.yaml (hunk 2 above)
Comment (pshirodkar-orcl) [reply platform-operator/helm_config/overrides/jaeger-operator-values.yaml:62]: I assume you meant the user as 1001. The jaeger-operator dockerfile specifically has a `USER` directive with user id as 1001 in the existing build from source.

File: platform-operator/helm_config/overrides/jaeger-operator-values.yaml (hunk 3 above)
Comment (pshirodkar-orcl) [reply platform-operator/helm_config/overrides/jaeger-operator-values.yaml:67]: I assume you meant the user as 1001. The jaeger-operator dockerfile specifically has a `USER` directive with user id as 1001 in the existing build from source.

File: platform-operator/helm_config/overrides/jaeger-operator-values.yaml (hunk 2 above)
Comment (pfmackin) [reply platform-operator/helm_config/overrides/jaeger-operator-values.yaml:62]: you need to look at the /etc/passwd entry for that user and see what group id is used.

Comment (pshirodkar-orcl) [reply platform-operator/helm_config/overrides/jaeger-operator-values.yaml:62]: The group is running as 65534. Output below:

<pre>
[~]$ kubectl -n verrazzano-monitoring exec -it jaeger-operator-6d67998b8d-v9lpb -- bash
bash-4.4$ whoami
whoami: cannot find name for user ID 1001
bash-4.4$ groups
nobody
bash-4.4$ cat /etc/passwd 
root:x:0:0:root:/root:/bin/bash
bin:x:1:1:bin:/bin:/sbin/nologin
daemon:x:2:2:daemon:/sbin:/sbin/nologin
adm:x:3:4:adm:/var/adm:/sbin/nologin
lp:x:4:7:lp:/var/spool/lpd:/sbin/nologin
sync:x:5:0:sync:/sbin:/bin/sync
shutdown:x:6:0:shutdown:/sbin:/sbin/shutdown
halt:x:7:0:halt:/sbin:/sbin/halt
mail:x:8:12:mail:/var/spool/mail:/sbin/nologin
operator:x:11:0:operator:/root:/sbin/nologin
games:x:12:100:games:/usr/games:/sbin/nologin
ftp:x:14:50:FTP User:/var/ftp:/sbin/nologin
nobody:x:65534:65534:Kernel Overflow User:/:/sbin/nologin
bash-4.4$
</pre>

**Review by pfmackin — APPROVED** [review]

**Pull request merged**